docs/example-confluence/src/confluenceParser.ts

```diff
@@ -0,0 +1,176 @@
+import * as TurndownService from 'turndown';
+
+interface HeadingData {
+    title: string;
+    hierarchy: string;
+    result: string;
+}
+
+interface HeadingStackItem {
+    level: number;
+    text: string;
+}
+
+export class ConfluenceDataParser {
+    private targetHeadingsLevel: number;
+    private pageName: string = '';
+    private turndownService: TurndownService;
+    private html: string;
+
+    /**
+     * Constructs a new ConfluenceDataParser instance.
+     * @param html The HTML content to parse
+     * @param pageName The name of the page
+     * @param targetHeadingsLevel The target heading level to extract
+     */
+    constructor(html: string, pageName: string, targetHeadingsLevel: number) {
+        this.html = html
+        this.targetHeadingsLevel = targetHeadingsLevel;
+        this.pageName = pageName;
+        this.initializeTurndownService();
+    }
+
+    /**
+     * Initializes the TurndownService with custom rules for Confluence content.
+     */
+    private initializeTurndownService(): void {
+        this.turndownService = new TurndownService({
+            headingStyle: 'atx',           // Use # ## ### format
+            bulletListMarker: '-',         // Use - for bullet points
+            emDelimiter: '_',              // Use _emphasis_
+            strongDelimiter: '**',         // Use **strong**
+            linkStyle: 'inlined',          // Use [text](url) format
+            linkReferenceStyle: 'full',    // Full reference links
+            blankReplacement: this.handlingForEmptyNode.bind(this)
+        });
+        this.addConfluenceRules();
+    }
+
+    /**
+     *  Adds custom rules to the TurndownService for handling Confluence specific elements.
+     */
+    private addConfluenceRules(): void {
+
+        // Handle Confluence task lists
+        this.turndownService.addRule('confluenceTaskList', {
+            filter: (node) => node.nodeName === 'AC:TASK-LIST',
```

Comment:
This suggestion is optional and is based on my own preference

```suggestion
            filter: ({ nodeName }) => nodeName === 'AC:TASK-LIST',
```